libcloud/storage/drivers/runabove.py

```diff
@@ -0,0 +1,289 @@
+        response = self.connection.request(action='%s/token' % API_ROOT)\
+            .object
+        self.auth_token = response["X-Auth-Token"]
+        self.auth_description = response['token']
+
+    def iterate_containers(self):
+        action = API_ROOT + '/storage'
+        response = self.connection.request(action)
+        return self._to_containers(response.object)
+
+    def iterate_container_objects(self, container):
+        action = '%s/storage/%s' % (API_ROOT, container.name)
+        location = container.extra['region']
+        data = {
+            'region': location,
+            'limit': 1000,
+        }
+        try:
+            response = self.connection.request(action, data=data)
+        except ProviderError as err:
+            if err.http_code == httplib.NOT_FOUND:
+                raise ContainerDoesNotExistError(value=None,
+                                                 driver=self,
+                                                 container_name=container.name)
+            raise err
+        json_response = response.object
+        container = self._to_container(json_response, location)
+        return self._to_objects(response.object['objects'],
+                                container=container)
+
+    def list_container_objects(self, container):
+        return list(self.iterate_container_objects(container))
+
+    def get_container(self, container_name, ex_location=None):
+        """
+        Return a container instance.
+
+        :param container_name: Container name.
+        :type container_name: ``str``
+
+        :param ex_location: Region where to get
+        :type ex_location: ``str``
+
+        :return: :class:`Container` instance.
+        :rtype: :class:`Container`
+        """
+        action = '%s/storage/%s' % (API_ROOT, container_name)
+        location = self._check_location(ex_location)
+        data = {
+            'region': location,
+        }
+        try:
+            response = self.connection.request(action, data=data)
+        except ProviderError as err:
+            if err.http_code == httplib.NOT_FOUND:
+                raise ContainerDoesNotExistError(value=None,
+                                                 driver=self,
+                                                 container_name=container_name)
+            raise err
+        return self._to_container(response.object, location=location)
+
+    def get_object(self, container_name, object_name, ex_location=None):
+        # TODO: Remake with Swift request
+        location = self._check_location(ex_location)
+        action = '%s/storage/%s' % (API_ROOT, container_name)
+        data = {
+            'region': location,
+            'limit': 1000,
+        }
+        try:
+            response = self.connection.request(action, data=data)
+            json_response = response.object
+            container = self._to_container(json_response, location)
+            obj = [o for o in json_response['objects']
```

Comment:
This late filtering could be very inefficient and slow if there are many objects...

Is it possible to do it differently (e.g. only retrieve details for a single object or perform some kind of name filtering)?